### ISSUE TITLE
Setting Environment Proxy (HTTP_PROXY etc.) breaks Some Tests

### ISSUE THREAD

**Issue by scirelli** [aio-libs/aiohttp#5554]
🐞 **Describe the bug**

Tests break when proxy environment variables (HTTP_PROXY, HTTPS_PROXY, etc.) are set.

💡 **To Reproduce**

```
export http_proxy='http://www.example.com,http://proxy.com'
make test
```
💡 **Expected behavior**

Tests should pass.

📋 **Logs/tracebacks**

```python-traceback (paste your traceback in the next line)
FAILED tests/test_helpers.py::test_proxies_from_env[https] - AssertionError: assert dict_keys(['http', 'https']) == {'https'}
FAILED tests/test_helpers.py::test_proxies_from_env[ws] - AssertionError: assert dict_keys(['http', 'ws']) == {'ws'}
FAILED tests/test_helpers.py::test_proxies_from_env[wss] - AssertionError: assert dict_keys(['http', 'wss']) == {'wss'}
FAILED tests/test_helpers.py::test_proxies_from_env_skipped[https] - AssertionError: assert {'http': ProxyInfo(proxy=URL('http://www.example.com,http//proxy.com'), proxy_auth=None)} == {}
FAILED tests/test_helpers.py::test_proxies_from_env_skipped[wss] - AssertionError: assert {'http': ProxyInfo(proxy=URL('http://www.example.com,http//proxy.com'), proxy_auth=None)} == {}
```

📋 **Your version of the Python**

```console
$ python --version
Python 3.8.7
```

📋 **Your version of the aiohttp/yarl/multidict distributions**

```console
$ python -m pip show aiohttp
Name: aiohttp
Version: 4.0.0a1
Summary: Async http client/server framework (asyncio)
Home-page: https://github.com/aio-libs/aiohttp
Author: Nikolay Kim
Author-email: fafhrd91@gmail.com
License: Apache 2
Location: /Users/cirelli/Projects/aiohttp
Requires: chardet, multidict, async-timeout, yarl, typing-extensions, frozenlist, aiosignal
Required-by: 
```
```console
$ python -m pip show multidict
Name: multidict
Version: 5.1.0
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /Users/cirelli/Projects/aiohttp/.venv/lib/python3.8/site-packages
Requires: 
Required-by: yarl, aiohttp
```
```console
$ python -m pip show yarl
Name: yarl
Version: 1.6.3
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /Users/cirelli/Projects/aiohttp/.venv/lib/python3.8/site-packages
Requires: idna, multidict
Required-by: aiohttp
```

📋 **Additional context**

Mac OS

**Comment by webknjaz** [comment]
That's unfortunate... Mind submitting a PR?

**Comment by scirelli** [comment]
Sure, I mentioned it in my last PR that I'd create a new one to fix it. I figured it required a new PR.

**Comment by webknjaz** [comment]
Ah, I somehow missed that.. Thanks!

**Comment by webknjaz** [comment]
Implemented by https://github.com/aio-libs/aiohttp/pull/5557 and https://github.com/aio-libs/aiohttp/pull/6184.
